lib/ansible/modules/windows/win_psmodule.ps1

```diff
@@ -117,15 +118,41 @@ Function Install-PsModule {
     else {      
       try{
         # Install NuGet Provider if needed
-        Install-NugetProvider -CheckMode $CheckMode
+        Install-NugetProvider -CheckMode $CheckMode;
+
+        # Need PSGet 1.6.0 for publishing and named repo usage
+        $psg = [PSCustomObject]@{ n = "PowerShellGet"; v = "1.6.0"};
```

Comment:
I'm not sure we should be uninstalling and updating PowerShellGet if it doesn't meet the minimum version. What we should do is;

* Document the minimum required version on the docs
* Fail in the module if the version does not meet the minimum requirement
* Provide an example in the docs that shows how to upgrade PowerShellGet that we can point it

Because the module currently works with the older version we should only get/throw the exception based on that minimum version if we are specifying the repository. This is probably more important as it sounds like `1.6.0` isn't builtin by default.

Also, any particular reason why you are using a custom object here?

Reply:
@jborean93 The custom object is merely syntactic sugar to make it apparent that the Name and Version of the module form a single requirement.

Installing PSG 1.6.0 does not uninstall the existing version, it installs side-by-side AFAIK. 

What do you think of making 1.6.0 the default minimum version, with the module installing it if necessary, along with documenting this?

Reply:
I personally dislike how win_chocolatey does that in the background, can you bring this up at the next Windows Working Group meeting to discuss as this should be a decision taken by the wider audience. You can add it to the agenda here https://github.com/ansible/community/issues/294.